src/main/java/hyperledger/besu/java/rest/client/utils/SmartContractUtility.java

```diff
@@ -15,93 +15,171 @@
+        .forEach(
+            abiPathSmartContractAddress ->
+                getAbiFileListForSmartContracts(
+                    baseAbiPath, abiPathSmartContractAddress, abiFileList));
     return abiFileList;
   }
 
-  private void getAbiListForSmartContracts(
-      String baseAbiPath, String parentFolderName, List<String> abiFileList) {
-    String smartContractPath = baseAbiPath + parentFolderName;
-    File[] abiFiles = new File(smartContractPath).listFiles();
-    Arrays.stream(abiFiles)
-        .forEach(file -> abiFileList.add(smartContractPath + "/" + file.getName()));
+  /**
+   * @param aboBasePath
+   * @param parentFolderNameForSCAddress
+   * @param abiPathFileList
+   */
+  private void getAbiFileListForSmartContracts(
+      final String aboBasePath,
+      final String parentFolderNameForSCAddress,
+      final List<String> abiPathFileList) {
+    String addressPath = aboBasePath + parentFolderNameForSCAddress;
+    try {
+      File[] abiFiles = new File(addressPath).listFiles();
+      Arrays.stream(abiFiles)
+          .forEach(
+              filePath -> {
+                abiPathFileList.add(addressPath + "/" + filePath.getName());
+              });
+    } catch (Exception e) {
+      log.error(ErrorCode.FILE_PARSE_ERROR.getReason(), e.getMessage());
+    }
   }
 
-  private void addEventNameByHash(String abiFilePath, Map<String, String> eventNameByHash) {
-    byte[] expectedTemplate;
+  /**
+   * @param abiPath
+   * @param eventNameHexMapping
+   */
+  private void addEventNameByHash(
+      final String abiPath, final Map<String, String> eventNameHexMapping)
+      throws UnsupportedEncodingException {
+    byte[] abiFileDefinition;
     try {
-      expectedTemplate = IOUtils.toByteArray(ResourceUtils.getURL(abiFilePath));
+      abiFileDefinition = IOUtils.toByteArray(ResourceUtils.getURL(abiPath));
     } catch (IOException e) {
       throw new FileParseException(ErrorCode.FILE_PARSE_ERROR, e.getMessage());
     }
-    String abiString = new String(expectedTemplate, StandardCharsets.UTF_8);
+    String abiDefinitionForSmartContractAddressStringified =
+        new String(abiFileDefinition, StandardCharsets.UTF_8);
     try {
-      mapper
+      OBJECT_MAPPER
           .readValue(
-              abiString,
-              new com.fasterxml.jackson.core.type.TypeReference<List<AbiDefinition>>() {})
+              abiDefinitionForSmartContractAddressStringified,
+              new TypeReference<List<AbiDefinition>>() { })
           .stream()
-          .forEach(abiDef -> createEventDefinition(abiDef, eventNameByHash));
+          .forEach(
+              abiDefinition -> {
+                createEventDefinition(abiDefinition, eventNameHexMapping);
+              });
 
     } catch (JsonProcessingException e) {
-      throw new InvalidArgumentException(ErrorCode.INVALID_ARGUMENT_FOUND, e.getMessage());
+      log.error("Unable to read Abi file", e.getMessage());
```

Comment:
Already throwing FileParseException in case of invalid abi file path. So, in the other cases like file parsing or invalid defs , simply logging the error message instead of throwing.